Content.Server/Lathe/LatheSystem.cs

```diff
@@ -406,6 +414,39 @@ protected override bool HasRecipe(EntityUid uid, LatheRecipePrototype recipe, La
             return GetAvailableRecipes(uid, component).Contains(recipe.ID);
         }
 
+        #region Temperature
+        public LatheTemperatureStatus GetLatheTemperatureStatus(EntityUid uid, LatheComponent component)
+        {
+            component.TempStatus = LatheTemperatureStatus.Normal;
+
+            if (component.MinTemp != null || component.MaxTemp != null)
+            {
+                var mix = _atmosphere.GetTileMixture(uid);
+
+                if (mix == null || mix.TotalMoles <= 0 || mix.Temperature <= component.MinTemp)
+                {
+                    component.TempStatus = LatheTemperatureStatus.Low;
+                }
+
+                else if (mix.Temperature >= component.MaxTemp)
+                {
+                    component.TempStatus = LatheTemperatureStatus.High;
+                }
+            }
+
+            UpdateTemperatureAppearance(uid, component.TempStatus);
+            return component.TempStatus;
+        }
+
+        private void UpdateTemperatureAppearance(EntityUid uid, LatheTemperatureStatus status, AppearanceComponent? appearance = null)
```

Comment:
```suggestion
        private void UpdateTemperatureAppearance(EntityUid uid, LatheTemperatureStatus status)
```
never used, so just remove the resolve